components/ds/ComboboxComponent.tsx

```diff
@@ -17,16 +17,19 @@ interface ComboboxProps {
   data: { value: string; label: string }[];
   onSelect(value: string): void;
   defaultValue?: string;
+  value?: string;
   disabled?: boolean;
 }
 
 export function Combobox(props: ComboboxProps) {
   const [open, setOpen] = useState(false);
-  const [value, setValue] = useState(props.defaultValue || "");
+  const [internalValue, setInternalValue] = useState(props.defaultValue || "");
+
+  const value = props.value !== undefined ? props.value : internalValue;
   const selectedItem = props.data.find((item) => item.value === value);
 
   const setNewValue = (value: string) => {
-    setValue(value);
+    setInternalValue(value);
     setOpen(false);
     props.onSelect(value);
   };
```

Comment:
Now, let's simplify this logic.
Since we always receive value as prop we can utilize it in our code via `props.value`, no need to reassign value to local variable, like this:

```
  const [open, setOpen] = useState(false);
  const selectedItem = props.data.find((item) => item.value === props.value);

  const setNewValue = (value: string) => {
    setOpen(false);
    props.onSelect(value);
  };
```

Reply:
You will also need to use props.value instead of value for `CheckIcon`

```
<CheckIcon
  className={cn(
    "ml-auto h-4 w-4",
    props.value === item.value ? "opacity-100" : "opacity-0"
  )}
/>
```